dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -625,6 +636,144 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
                 "Removing bundles in a separated process, the result of the operation will be notified")).build();
     } // deleteAllSuccess.
 
+
+    @Path("/_download/{bundleId}")
+    @GET
+    @JSONP
+    @NoCache
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public final Response downloadBundle(@Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            @PathParam("bundleId") final String bundleId) {
+
+        final InitDataObject initData =
+                new WebResource.InitBuilder(webResource)
+                        .requiredBackendUser(true)
+                        .requestAndResponse(request, response)
+                        .rejectWhenNoUser(true)
+                        .requiredPortlet("publishing-queue")
+                        .init();
+        try {
+            final Bundle bundle = APILocator.getBundleAPI().getBundleById(bundleId);
+            if (!UtilMethods.isSet(bundle)) {
+                throw new DoesNotExistException("Bundle with ID: " + bundleId + " not found");
+            }
+            final File bundleFile = new File(
+                    ConfigUtils.getBundlePath() + File.separator + bundle.getId() + ".tar.gz");
+            if (!bundleFile.exists()) {
+                throw new DoesNotExistException(
+                        "The bundle has not been generated for the provided bundle ID: "
+                                + bundleId);
+            }
+            final String bundleName = bundle.getName().replaceAll("[^\\w.-]", "_");
+            response.setHeader( "Content-Disposition", "attachment; filename=" +bundleName  +"-"+ bundle.getId() + ".tar.gz" );
+            return Response.ok(bundleFile, "application/x-tgz").build();
+        }catch (DoesNotExistException e){
+            Logger.error(this,e.getMessage());
+            return ExceptionMapperUtil.createResponse("",e.getMessage(),Response.Status.NOT_FOUND);
```

Comment:
what would happen if this response is sent considering that the endpoint is tagged with `@Produces(MediaType.APPLICATION_OCTET_STREAM)` ?

Reply:
It response the error like this:
```
{
    "message": "Bundle with ID: bundle-id-not-exist not found"
}
```
<img width="1383" alt="Screen Shot 2020-10-07 at 8 49 33 AM" src="https://user-images.githubusercontent.com/5278970/95347362-07832680-087a-11eb-9951-c40b2e17a465.png">